### PR TITLE
refactor(nuxt): prefix all core modules with `nuxt:`

### LINKED ISSUE #29965: Inadvertently overriding a built-in nuxt module should warn the user
### Environment

Warn users if they create a module the same as a built-in Nuxt module.

### Reproduction

1. Create a module
2. Name it `components`
3. It will override the built-in `components` module

### Describe the bug

Overriding the default built-in module (something like `components`) should be allowed; however, it's not obvious that this will likely completely break the app. 

Nuxt should at least warn in the logs that a built-in is being overridden, etc.

### Additional context

This was demonstrated (accidentally) at Vue Toronto

### Logs

```shell-script

```

### PR DESCRIPTION
**Pull request by huang-julien**
### 🔗 Linked issue
resolves #29965

### 📚 Description

This PR prefix all core modules name with `nuxt:` to harmonize then naming of core modules with core plugins.

This is kinda a breaking change if users were overwriting core modules ? 

## Release Notes

- **New Features**
	- Updated module naming conventions for clarity (e.g., `nuxt:components`, `nuxt:imports`, `nuxt:pages`).
	- Enhanced handling of component directories and router options for improved functionality.
	- Introduced new type templates for middleware and layouts to enhance type safety.
	- Improved dynamic updates for imports and better handling of component paths.

- **Bug Fixes**
	- Improved error handling and logging for schema loading and route parsing.

- **Documentation**
	- Enhanced comments and structure to clarify the integration of modules within the Nuxt framework. 

These changes aim to improve the overall organization, error handling, and functionality of the Nuxt framework for developers.

### CODE REVIEW

File: packages/nuxt/src/pages/module.ts [hunk 1]
```diff
@@ -23,7 +23,7 @@ const OPTIONAL_PARAM_RE = /^\/?:.*(?:\?|\(\.\*\)\*)$/
 
 export default defineNuxtModule({
   meta: {
-    name: 'pages',
+    name: 'nuxt:pages',
```
Comment (danielroe) [inline packages/nuxt/src/pages/module.ts:26]:
```suggestion
    name: 'nuxt:pages',
    configKey: 'pages',
```

File: packages/nuxt/src/core/schema.ts [hunk 2]
```diff
@@ -16,7 +16,7 @@ import { createJiti } from 'jiti'
 
 export default defineNuxtModule({
   meta: {
-    name: 'nuxt-config-schema',
+    name: 'nuxt:nuxt-config-schema',
```
[the remark this answers is not in the document]
Comment (danielroe) [reply packages/nuxt/src/core/schema.ts:19]: cc: @huang-julien - this is a good spot from 🐇 - we need to update the logic there, not just for this one module name but for all of them (could even simplify to a `startsWith`)

File: packages/nuxt/src/utils.ts [hunk 3]
```diff
@@ -8,3 +8,5 @@ export function toArray<T> (value: T | T[]): T[] {
 export async function isDirectory (path: string) {
   return (await fsp.lstat(path)).isDirectory()
 }
+
+export const NUXT_PREFIX = 'nuxt:'
```
Comment (huang-julien) [inline packages/nuxt/src/utils.ts:12]: for better naming consistency, do you think we should use it everywhere (plugins + modules) to define names ?
`name: NUXT_PREFIX + 'page'`

Comment (danielroe) [reply packages/nuxt/src/utils.ts:12]: I think just using the string should be ok

**Review by danielroe — APPROVED** [review]

**Review by danielroe — APPROVED** [review]

**Pull request merged**